src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/PackageRestoreDataSource.cs

```diff
@@ -101,11 +144,14 @@ internal class PackageRestoreDataSource : ChainedProjectValueDataSourceBase<Rest
 
         internal async Task<IEnumerable<IProjectVersionedValue<RestoreData>>> RestoreAsync(IProjectVersionedValue<PackageRestoreUnconfiguredInput> e)
         {
-            // No configurations - likely during project close
-            if (!_enabled || e.Value.RestoreInfo is null)
+            // No configurations - likely during project close.
+            // Check if out of date to prevent extra restore under some conditions.
+            if (!_enabled || e.Value.RestoreInfo is null || IsProjectConfigurationVersionOutOfDate(e.Value.ConfiguredInputs) is false)
```

Comment:
"IsProjectConfigurationVersionOutOfDate(e.Value.ConfiguredInputs) is false" means "if not out of date" - this seems wrong?